nmigen/lib/scheduler.py

```diff
@@ -0,0 +1,60 @@
+from .. import *
+
+
+__all__ = ["RoundRobin"]
+
+
+class RoundRobin(Elaboratable):
+    """Round-robin scheduler.
+
+    For a given set of requests, the round-robin scheduler will
+    grant one request. Once it grants a request, if any other
+    requests are active, it grants the next active request with
+    a greater number, restarting from zero once it reaches the
+    highest one.
+
+    Use :class:`EnableInserter` to control when the scheduler
+    is updated.
+
+    Parameters
+    ----------
+    width : int
```

Comment:
Ah, one last bikeshed. Talking about the "width" of a scheduler doesn't make much sense to me. Perhaps "size" or "count"?